src/Storefront/Resources/app/storefront/src/component-system/README.md

```diff
@@ -0,0 +1,361 @@
+
+You can build the options individually from your Twig component properties and other data, or simply pass through the component properties to your JavaScript component.
+
+```twig
+{# views/component/MyComponent.html.twig #}
+
+{% props
+    foo = "bar",
+    custom = true,
+%}
+
+<div data-my-component 
+     data-my-component-options='{{ this.props|json_encode }}'>
+
+    {# Some component logic ... #}
+</div>
+```
+
+
+## Component Communication
+
+### Event System
+
+To react to actions from other components, there is a new central event system available which can be accessed via the global `window.Shopware` singleton.
+
+In your component you can emit events to inform others about an action and pass additional data via the event.
+
+```javascript
+// MyComponent.js
+
+class MyComponent extends ShopwareComponent {
+
+    // ...
+
+    doSomething() {
+        const message = 'Hello World!';
+
+        window.Shopware.emit('MyComponent:DoSomething', message);
+    }
+}
+```
+
+Other components can the subscribe to this event to react on that.
+
+```javascript
+// SomeOtherComponent.js
+
+class SomeOtherComponent extends ShopwareComponent {
+
+    init() {
+        window.Shopware.on('MyComponent:DoSomething', (message) => {
+            this.el.innerText = message;
+        });
+    }
+}
+```
+
+Of course, you can also register to events from anywhere else, also from outside of the component system. For example, if you just want to extend the logic of an existing component.
+
+### Event Interception
+
+In addition to the normal asynchronous events, there is a seprate event type which expects a return value that gets further processed within the component. These events make it even easier to extend a components logic and offers a bunch of different use cases, like manipulating request data before it gets send. 
+
+For example the BuyButton component offers an event `BuyButton:PreSubmit` which is interceptable, as it is called via `emitInterception()`. It is triggered when a user clicks the buy button of a product.
+
+```javascript
+// BuyButton.js
+
+class BuyButton extends ShopwareComponent {
+
+    // ...
+
+    onFormSubmit(event) {
+        event.preventDefault();
+
+        let requestUrl = this.el.getAttribute('action');
+        let formData = window.Shopware.serializeForm(this.el);
+
+        [ requestUrl, formData ] = window.Shopware.emitInterception('BuyButton:PreSubmit', requestUrl, formData);
+
+        window.Shopware.emit('BuyButton:Submit', requestUrl, formData);
+
+        window.Shopware.callPluginMethod('OffCanvasCart', 'openOffCanvas', requestUrl, formData);
+    }
+}
+```
+
+You can see that the event `BuyButton:PreSubmit` offers the opportunity to manipulate the `formData` before it gets sent. From any other script you can intercept this event and work with the parameters from the event.
+
+```javascript
+// Intercept the buy button event
+window.Shopware.intercept('BuyButton:PreSubmit', (requestUrl, formData) => {
```

Comment:
Should we call this `onIntercept` to have a similar naming scheme with "on"? To me this makes it a bit clearer that this is the "listener/subscriber".

`emit`, `on`, `emitIntercept`, `onIntercept`.